crates/polars-core/src/serde/chunked_array.rs

```diff
@@ -144,7 +159,7 @@ impl Serialize for CategoricalChunked {
         S: Serializer,
     {
         {
-            let mut state = serializer.serialize_map(Some(3))?;
+            let mut state = serializer.serialize_map(Some(4))?;
```

Comment:
Was this a bug? If so, can you add a failing test case for this? Perhaps we shouldn't only test round-trip but actually the output strings as well.

Reply:
This one is already fixed in main.